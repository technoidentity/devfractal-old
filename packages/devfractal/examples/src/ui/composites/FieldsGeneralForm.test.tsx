import React from 'react'
import renderer, { ReactTestRenderer } from 'react-test-renderer'
import { FieldsGeneralForm } from './FieldsGeneralForm'

test.skip('Fields General Form', () => {
  const component: ReactTestRenderer = renderer.create(<FieldsGeneralForm />)
  expect(component.toJSON()).toMatchSnapshot()
})
