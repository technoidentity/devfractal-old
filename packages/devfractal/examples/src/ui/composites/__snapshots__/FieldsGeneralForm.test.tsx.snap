// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fields General Form 1`] = `
<section
  className="section"
>
  <div
    className="container"
  >
    <form
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <div
        className="field"
      >
        <label
          className="label"
        >
          Name
        </label>
        <div
          className="control"
        >
          <input
            className="input"
            name="name"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Input Text"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="field"
      >
        <label
          className="label"
        >
          Username
        </label>
        <div
          className="control"
        >
          <input
            className="input"
            name="username"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Input Text"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        className="field"
      >
        <label
          className="label"
        >
          Email
        </label>
        <div
          className="control"
        >
          <input
            className="input"
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Email Input"
            type="email"
            value=""
          />
        </div>
      </div>
      <div
        className="field"
      >
        <label
          className="label"
        >
          Subject
        </label>
        <div
          className="control"
        >
          <div
            className="select"
          >
            <select
              className=""
              name="select"
              onBlur={[Function]}
              onChange={[Function]}
              value=""
            >
              <option
                value="dropdown"
              >
                Select dropdown
              </option>
              <option
                value="options"
              >
                With options
              </option>
            </select>
          </div>
        </div>
      </div>
      <div
        className="field"
      >
        <label
          className="label"
        >
          Message
        </label>
        <div
          className="control"
        >
          <textarea
            className="textarea"
            name="message"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Textarea"
            rows={4}
            value=""
          />
        </div>
      </div>
      <div
        className="field"
      >
        <div
          className="control"
        >
          <label
            className="checkbox"
          >
            <input
              checked={false}
              className="checkbox"
              name="agree"
              onBlur={[Function]}
              onChange={[Function]}
              type="checkbox"
              value={false}
            />
             
            I agree to the
            <a
              href="#!"
            >
               terms and conditions
            </a>
          </label>
        </div>
      </div>
      <div
        className="field is-grouped"
      >
        <div
          className="control"
        >
          <label
            className="radio"
          >
            <input
              checked={false}
              name="radio"
              onBlur={[Function]}
              onChange={[Function]}
              type="radio"
              value="yes"
            />
             Yes
          </label>
          <label
            className="radio"
          >
            <input
              checked={true}
              name="radio"
              onBlur={[Function]}
              onChange={[Function]}
              type="radio"
              value="no"
            />
             No 
          </label>
        </div>
      </div>
      <div
        className="field is-grouped is-grouped-centered"
      >
        <div
          className="control"
        >
          <button
            className="button is-info"
            type="submit"
          >
            Submit
          </button>
        </div>
        <div
          className="control"
        >
          <button
            className="button is-danger"
            type="reset"
          >
            Reset
          </button>
        </div>
      </div>
      <pre>
        {
  "name": "",
  "username": "",
  "email": "",
  "select": "",
  "message": "",
  "agree": false,
  "radio": "no"
}
      </pre>
    </form>
  </div>
</section>
`;
